### ISSUE TITLE
[Bug] UI elements do not scale properly on Hi-DPI display

### ISSUE THREAD

**Issue by Explorare** [pit-ray/win-vind#139]
# About Bug
UI elements do not scale properly as intended on a Hi-DPI screen. 
In my case, I'm using a tablet with 3000*2000 13 inch display, 200% scale system-wide. 
While using "application" scale method, the UI elements are rendered in correct places, but too small, it should be scaled to 200%.
While using "system (enhanced)" method, all UI elements are rendered in 1/4 up-left part of the screen, and small. 
There is a closed issue for the same problem. 
[\[Bug\] Easy click cannot work on scaled monitor · Issue #54 · pit-ray/win-vind](https://github.com/pit-ray/win-vind/issues/54)

![image](https://user-images.githubusercontent.com/5090766/215317104-fa3f3be4-78c5-4369-bbab-5b688637fa23.png)

# Environments

|Environments|Info|
|:---:|:---|
|Windows Edition| 10 Enterprise|
|Windows Version|22H2|
|Windows Build|19045.2486|
|win-vind Version|5.0.0 beta5|
|win-vind Install Type|zip|

**Comment by pit-ray** [comment]
Thanks for participating in the beta test.

I fixed the position correction for high DPI in previous #54, but not the scaling yet.

So I fixed this time for scaling, please try the following commit for a corrected version that automatically scales the text size according to DPI. Attached is the build of the commit.
[win-vind_issue139_64bit_portable.zip](https://github.com/pit-ray/win-vind/files/10529633/win-vind_issue139_64bit_portable.zip)

**Comment by Explorare** [comment]
Thanks for your quick response. I've tested this build and the UI scale works as intend out-of-box. But the program will crash when right-click the tray icon for menu.

Here is the debug trace from WinDbg
```
0:000> g
ModLoad: 00007ffd`f12f0000 00007ffd`f1322000   C:\WINDOWS\System32\IMM32.DLL
ModLoad: 00007ffd`ee780000 00007ffd`ee792000   C:\WINDOWS\SYSTEM32\kernel.appcore.dll
ModLoad: 00007ffd`f0940000 00007ffd`f09c2000   C:\WINDOWS\System32\bcryptPrimitives.dll
ModLoad: 00007ffd`f2d80000 00007ffd`f2e2f000   C:\WINDOWS\System32\clbcatq.dll
ModLoad: 00007ffd`c94f0000 00007ffd`c97e5000   C:\Windows\System32\uiautomationcore.dll
ModLoad: 00007ffd`f2cb0000 00007ffd`f2d7d000   C:\WINDOWS\System32\OLEAUT32.dll
ModLoad: 00007ffd`e91f0000 00007ffd`e92e6000   C:\Windows\System32\PROPSYS.dll
ModLoad: 00007ffd`f0750000 00007ffd`f07fb000   C:\WINDOWS\SYSTEM32\sxs.dll
mincore\com\oleaut32\dispatch\ups.cpp(2122)\OLEAUT32.dll!00007FFDF2CB9DD6: (caller: 00007FFDF2CB91E9) ReturnHr(1) tid(35ec) 8002801D Library not registered.
mincore\com\oleaut32\dispatch\ups.cpp(2122)\OLEAUT32.dll!00007FFDF2CB9DD6: (caller: 00007FFDF2CB91E9) ReturnHr(2) tid(35ec) 8002801D Library not registered.
ModLoad: 00007ffd`f28e0000 00007ffd`f29f5000   C:\WINDOWS\System32\MSCTF.dll
ModLoad: 00007ffd`ee9a0000 00007ffd`ef132000   C:\WINDOWS\SYSTEM32\windows.storage.dll
ModLoad: 00007ffd`f02c0000 00007ffd`f02f0000   C:\WINDOWS\SYSTEM32\Wldp.dll
ModLoad: 00007ffd`da770000 00007ffd`da81c000   C:\WINDOWS\SYSTEM32\TextShaping.dll
ModLoad: 00007ffd`eaa60000 00007ffd`eab52000   C:\Windows\System32\CoreMessaging.dll
ModLoad: 00007ffd`f2be0000 00007ffd`f2c4b000   C:\WINDOWS\System32\WS2_32.dll
ModLoad: 00007ffd`ea380000 00007ffd`ea6de000   C:\Windows\System32\CoreUIComponents.dll
ModLoad: 00007ffd`efa90000 00007ffd`efac3000   C:\WINDOWS\SYSTEM32\ntmarta.dll
ModLoad: 00000137`ee640000 00000137`ee794000   C:\WINDOWS\SYSTEM32\wintypes.dll
ModLoad: 00007ffd`e93a0000 00007ffd`e94f4000   C:\WINDOWS\SYSTEM32\wintypes.dll
ModLoad: 00000137`ee7a0000 00000137`ee8f4000   C:\WINDOWS\SYSTEM32\wintypes.dll
ModLoad: 00007ffd`ba500000 00007ffd`ba5a1000   C:\Windows\System32\UiaManager.dll
ModLoad: 00007ffd`d46a0000 00007ffd`d4706000   C:\Windows\System32\OLEACC.dll
(19a8.b98): C++ EH exception - code e06d7363 (first chance)
ModLoad: 00007ffd`da400000 00007ffd`da40d000   C:\WINDOWS\SYSTEM32\LINKINFO.dll
ModLoad: 00007ffd`f1120000 00007ffd`f116e000   C:\WINDOWS\System32\CFGMGR32.dll
ModLoad: 00007ffd`f0880000 00007ffd`f089f000   C:\WINDOWS\SYSTEM32\profapi.dll
ModLoad: 00007ffd`eac90000 00007ffd`ead21000   C:\WINDOWS\SYSTEM32\apphelp.dll
ModLoad: 00007ffd`cf690000 00007ffd`cf720000   C:\Windows\System32\appresolver.dll
ModLoad: 00007ffd`e2140000 00007ffd`e219b000   C:\Windows\System32\Bcp47Langs.dll
ModLoad: 00007ffd`ef420000 00007ffd`ef449000   C:\Windows\System32\SLC.dll
ModLoad: 00007ffd`ef3f0000 00007ffd`ef415000   C:\Windows\System32\sppc.dll
(19a8.35ec): C++ EH exception - code e06d7363 (first chance)
(19a8.35ec): Unknown exception - code c000041d (!!! second chance !!!)
*** WARNING: Unable to verify checksum for C:\Users\explo\Desktop\win-vind\win-vind.exe
KERNELBASE!RaiseException+0x69:
00007ffd`f0d2cd29 0f1f440000      nop     dword ptr [rax+rax]
```

**Comment by pit-ray** [comment]
It seems to be a dependency issue at build time, so I tried building it on another computer, Please try it again.
[win-vind_issue139.2_64bit_portable.zip](https://github.com/pit-ray/win-vind/files/10529783/win-vind_issue139.2_64bit_portable.zip)

**Comment by Explorare** [comment]
Still have the same issue. 
```
Microsoft (R) Windows Debugger Version 10.0.25200.1003 AMD64
Copyright (c) Microsoft Corporation. All rights reserved.

CommandLine: C:\Users\explo\Desktop\win-vind\win-vind.exe

************* Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*
Symbol search path is: srv*
Executable search path is: 
ModLoad: 00007ff7`f74e0000 00007ff7`f7d1a000   image00007ff7`f74e0000
ModLoad: 00007ffd`f3250000 00007ffd`f3448000   ntdll.dll
ModLoad: 00007ffd`f2250000 00007ffd`f230f000   C:\WINDOWS\System32\KERNEL32.DLL
ModLoad: 00007ffd`f0d00000 00007ffd`f0fd2000   C:\WINDOWS\System32\KERNELBASE.dll
ModLoad: 00007ffd`f2a10000 00007ffd`f2b35000   C:\WINDOWS\System32\RPCRT4.dll
ModLoad: 00007ffd`d9110000 00007ffd`d93aa000   C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.19041.1110_none_60b5254171f9507e\COMCTL32.dll
ModLoad: 00007ffd`edee0000 00007ffd`edf7e000   C:\WINDOWS\SYSTEM32\UxTheme.dll
ModLoad: 00007ffd`f2c50000 00007ffd`f2ca5000   C:\WINDOWS\System32\SHLWAPI.dll
ModLoad: 00007ffd`f2b40000 00007ffd`f2bde000   C:\WINDOWS\System32\msvcrt.dll
ModLoad: 00007ffd`f2310000 00007ffd`f2665000   C:\WINDOWS\System32\combase.dll
ModLoad: 00007ffd`f2f40000 00007ffd`f2f6b000   C:\WINDOWS\System32\GDI32.dll
ModLoad: 00007ffd`f2680000 00007ffd`f2821000   C:\WINDOWS\System32\USER32.dll
ModLoad: 00007ffd`f0b30000 00007ffd`f0c30000   C:\WINDOWS\System32\ucrtbase.dll
ModLoad: 00007ffd`f0cd0000 00007ffd`f0cf2000   C:\WINDOWS\System32\win32u.dll
ModLoad: 0000025d`a06d0000 0000025d`a06f2000   C:\WINDOWS\System32\win32u.dll
ModLoad: 00007ffd`f1010000 00007ffd`f111f000   C:\WINDOWS\System32\gdi32full.dll
ModLoad: 00007ffd`ee670000 00007ffd`ee69f000   C:\WINDOWS\SYSTEM32\dwmapi.dll
ModLoad: 00007ffd`f1330000 00007ffd`f1a74000   C:\WINDOWS\System32\SHELL32.dll
ModLoad: 00007ffd`f0c30000 00007ffd`f0ccd000   C:\WINDOWS\System32\msvcp_win.dll
ModLoad: 00007ffd`f0800000 00007ffd`f082e000   C:\WINDOWS\SYSTEM32\USERENV.dll
ModLoad: 0000025d`a06d0000 0000025d`a076d000   C:\WINDOWS\System32\msvcp_win.dll
ModLoad: 00007ffd`f30e0000 00007ffd`f320a000   C:\WINDOWS\System32\ole32.dll
ModLoad: 0000025d`a06d0000 0000025d`a06d9000   C:\WINDOWS\SYSTEM32\icuuc.dll
ModLoad: 00007ffd`f20b0000 00007ffd`f218a000   C:\WINDOWS\System32\COMDLG32.dll
ModLoad: 00007ffd`d4830000 00007ffd`d48c8000   C:\WINDOWS\SYSTEM32\WINSPOOL.DRV
ModLoad: 00007ffd`f2e90000 00007ffd`f2f3d000   C:\WINDOWS\System32\shcore.dll
ModLoad: 00007ffd`f2830000 00007ffd`f28de000   C:\WINDOWS\System32\ADVAPI32.dll
ModLoad: 00007ffd`f21b0000 00007ffd`f224c000   C:\WINDOWS\System32\sechost.dll
ModLoad: 00007ffd`d48d0000 00007ffd`d48d7000   C:\WINDOWS\SYSTEM32\MSIMG32.dll
ModLoad: 0000025d`a06d0000 0000025d`a06d9000   C:\WINDOWS\SYSTEM32\icuuc.dll
ModLoad: 00007ffd`cd820000 00007ffd`cda4e000   C:\WINDOWS\SYSTEM32\icu.DLL
(1994.3f1c): Break instruction exception - code 80000003 (first chance)
ntdll!LdrpDoDebuggerBreak+0x30:
00007ffd`f3320950 cc              int     3
0:000> g
ModLoad: 00007ffd`f12f0000 00007ffd`f1322000   C:\WINDOWS\System32\IMM32.DLL
ModLoad: 00007ffd`ee780000 00007ffd`ee792000   C:\WINDOWS\SYSTEM32\kernel.appcore.dll
ModLoad: 00007ffd`f0940000 00007ffd`f09c2000   C:\WINDOWS\System32\bcryptPrimitives.dll
ModLoad: 00007ffd`f2d80000 00007ffd`f2e2f000   C:\WINDOWS\System32\clbcatq.dll
ModLoad: 00007ffd`c94f0000 00007ffd`c97e5000   C:\Windows\System32\uiautomationcore.dll
ModLoad: 00007ffd`f2cb0000 00007ffd`f2d7d000   C:\WINDOWS\System32\OLEAUT32.dll
ModLoad: 00007ffd`e91f0000 00007ffd`e92e6000   C:\Windows\System32\PROPSYS.dll
ModLoad: 00007ffd`f0750000 00007ffd`f07fb000   C:\WINDOWS\SYSTEM32\sxs.dll
mincore\com\oleaut32\dispatch\ups.cpp(2122)\OLEAUT32.dll!00007FFDF2CB9DD6: (caller: 00007FFDF2CB91E9) ReturnHr(1) tid(3f1c) 8002801D Library not registered.
mincore\com\oleaut32\dispatch\ups.cpp(2122)\OLEAUT32.dll!00007FFDF2CB9DD6: (caller: 00007FFDF2CB91E9) ReturnHr(2) tid(3f1c) 8002801D Library not registered.
ModLoad: 00007ffd`f28e0000 00007ffd`f29f5000   C:\WINDOWS\System32\MSCTF.dll
ModLoad: 00007ffd`ee9a0000 00007ffd`ef132000   C:\WINDOWS\SYSTEM32\windows.storage.dll
ModLoad: 00007ffd`f02c0000 00007ffd`f02f0000   C:\WINDOWS\SYSTEM32\Wldp.dll
ModLoad: 00007ffd`da770000 00007ffd`da81c000   C:\WINDOWS\SYSTEM32\TextShaping.dll
ModLoad: 00007ffd`da400000 00007ffd`da40d000   C:\WINDOWS\SYSTEM32\LINKINFO.dll
ModLoad: 00007ffd`f1120000 00007ffd`f116e000   C:\WINDOWS\System32\CFGMGR32.dll
ModLoad: 00007ffd`f0880000 00007ffd`f089f000   C:\WINDOWS\SYSTEM32\profapi.dll
ModLoad: 00007ffd`eac90000 00007ffd`ead21000   C:\WINDOWS\SYSTEM32\apphelp.dll
ModLoad: 00007ffd`cf690000 00007ffd`cf720000   C:\Windows\System32\appresolver.dll
ModLoad: 00007ffd`ef420000 00007ffd`ef449000   C:\Windows\System32\SLC.dll
ModLoad: 00007ffd`e2140000 00007ffd`e219b000   C:\Windows\System32\Bcp47Langs.dll
ModLoad: 00007ffd`ef3f0000 00007ffd`ef415000   C:\Windows\System32\sppc.dll
(1994.3f1c): C++ EH exception - code e06d7363 (first chance)
(1994.3f1c): Unknown exception - code c000041d (!!! second chance !!!)
KERNELBASE!RaiseException+0x69:
00007ffd`f0d2cd29 0f1f440000      nop     dword ptr [rax+rax]
```

**Comment by pit-ray** [comment]
I understand.
I try to reproduce the same environment with Virtual box.

**Comment by pit-ray** [comment]
@Explorare 
I have not yet been able to reproduce it.
Are there any errors output to the log directory?

**Comment by Explorare** [comment]
Sorry for the late reply. 
Yes, I have tried on another laptop with the same OS but much simpler environment, and it works as intend, no crash, and scale works fine. I've also tried to set the scale to 100% on the tablet but the program still crash. So it is not related to scale function. 
v4.4.3 also began to behave the same way, but not 2 days ago. There must be some changes on my system cause the issue. The log didn't catch the crash. 
```
========== System Infomation ==========
[Windows]
      Edition: Windows 10 Enterprise
      Version: 22H2
Build Numbers: 10.0.19045
 Architecture: x64

[win-vind]
      Version: 5.0.0.5
=======================================
```
The GUI scale support has been fixed, so I will close this issue.
I will try to find out what cause the crash, and open another issue for it if needed.
Thanks for your time and effort. Have a good day.
